### PR TITLE
feat(integrations): add support for garmin

### PR DESCRIPTION
**Pull request by hassan254-prog**
## Describe your changes

- Made some platform changes to accommodate this.
- Add support for Garmin

## Issue ticket number and link
[EXT-86](https://linear.app/nango/issue/EXT-86/add-support-for-garmin-api)
## Test
Tested Garmin and the rest of providers using `Oauth1` to ensure we don't break.

### CODE REVIEW

**Comment by henrymgarrett** [comment]
Thanks so much for all of the hard work on this @hassan254-prog !

**Comment by hassan254-prog** [comment]
@henrymgarrett You're welcome! It was a team effort, and I'm glad we could get it done. Looking forward to more collaborations in the future!

File: packages/server/lib/clients/oauth1.client.ts [hunk 1]
```diff
@@ -107,10 +107,11 @@ export class OAuth1Client {
...
 
-    getAuthorizationURL(requestToken: OAuth1RequestTokenResult) {
+    getAuthorizationURL(requestToken: OAuth1RequestTokenResult, oAuth1CallbackURL: string) {
         const scopes = this.config.oauth_scopes ? this.config.oauth_scopes.split(',').join(this.authConfig.scope_separator || ' ') : '';
 
-        let additionalAuthParams = {};
+        let additionalAuthParams: Record<string, any> = {};
+        additionalAuthParams['oauth_callback'] = oAuth1CallbackURL;
         if (this.authConfig.authorization_params) {
             additionalAuthParams = this.authConfig.authorization_params;
```
Comment (bodinsamuel) [inline packages/server/lib/clients/oauth1.client.ts:115]: is it okay if this override the whole object then?

File: packages/server/lib/clients/oauth1.client.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ export class OAuth1Client {
             typeof this.authConfig.token_url === 'string' ? this.authConfig.token_url : (this.authConfig.token_url?.['OAUTH1'] as string),
             this.config.oauth_client_id,
             this.config.oauth_client_secret,
-            '1.0A',
+            '1.0',
```
Comment (bodinsamuel) [inline packages/server/lib/clients/oauth1.client.ts:33]: Can you explain what does it changes?

**Review by bodinsamuel — APPROVED** [review]

**Review by khaliqgant — APPROVED** [review]

**Pull request merged**